docs/plugins/npe2_getting_started.md

```diff
@@ -0,0 +1,281 @@
+should conform to the [PEP8][] naming convention.
+
+When the cookiecutter asked to include a reader plugin, we selected "y", and
+in the next question we told cookiecutter that our reader should be invoked
+for files matching the '\*.npy' glob pattern.
+
+After answering all the prompts, `cookiecutter` will create a directory called
+`my_npy_reader` in the current directory that holds the generated files. It
+will look something like this:
+
+```
+my-npy-reader
+├── LICENSE
+├── MANIFEST.in
+├── README.md
+├── docs
+│   └── index.md
+├── mkdocs.yml
+├── requirements.txt
+├── setup.cfg                  <-- Defines an entry point group
+├── setup.py                       pointing to the plugin manifest
+├── src
+│   └── my_npy_reader
+│       ├── __init__.py
+│       ├── _reader.py         <-- example reader code
+│       ├── _tests
+│       │   ├── __init__.py
+│       │   └── test_reader.py
+│       └── napari.yml         <-- The plugin manifest
+└── tox.ini
+```
+
+Inside `setup.cfg` we added an entry point group that is used to identify
+`npe2`-style plugins and to locate the plugin manifest file, `napari.yml`.
+
+```cfg
+[options.entry_points]
+napari.manifest =
+    my-npy-reader = my_npy_reader:napari.yaml
+```
+
+The plugin manifest file is specified relative to the submodule root path. For
+the example it will be loaded from:
+`<path/to/my-npy-reader>/my_npy_reader/src/my_npy_reader/napari.yaml`.
+
+The generated `napari.yml` file looks like this:
+
+```yaml
+name: my-npy-reader
+author: Hero Protagonist
+display_name: My Plugin
+entry_point: my_npy_reader
+
+contributions:
+  commands:
+    - id: my_npy_reader.get_reader
+      python_name: my_plugin._reader:napari_get_reader
+      title: Open data with napari FooBar
+  readers:
+    - command: my_npy_reader.get_reader
+      accepts_directories: false
+      filename_patterns: ["*.npy"]
+```
+
+With `npe2` installed, we can check that this is a valid plugin manifest:
+
+```bash
+> npe2 validate src/my_npy_reader
+✔ Manifest for 'napari FooBar' valid!
+```
+
+See the {ref}`npe2-manifest-spec` for more details.
+
+## 2. Write code to implement your plugin
+
+For our example `*.npy` reader, edit `src/my_npy_reader/_reader.py`:
+
+```python
+import numpy as np
+
+def npy_file_reader(path):
+    array = np.load(path)
+    # return it as a list of LayerData tuples,
+    # here with no optional metadata
+    return [(array,)]
+
+# This is the function referenced by the reader command in `napari.yml`.
+def napari_get_reader(path):
+    # remember, path can be a list, so we check its type first...
+    # (this example plugin doesn't handle lists)
+    if isinstance(path, str) and path.endswith(".npy"):
+        # If we recognize the format, we return the actual reader function
+        return npy_file_reader
+    # otherwise we return None.
+    return None
+```
+
+You can start testing your plugin using a local editable install, and opening
+napari:
+
+```bash
+pip install -e .
+napari
```

Comment:
Can we update this to be a single command that "executes" the plugin? e.g. `napari path/to/file.npy`? That's not a super interesting example, but I always have to look up the equivalent command for pre-loading a plugin widget.